platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/webview/WebViewCapture.kt

```diff
@@ -0,0 +1,258 @@
+            buildMap {
+                put("_url", url)
+                if (isError) {
+                    errorCode?.let { put("_errorCode", it) }
+                    errorCodeName?.let { put("_errorCodeName", it) }
+                    errorDescription?.let { put("_description", it) }
+                }
+            }
+        }
+    }
+    /**
+     * Companion object for WebViewCapture.
+     */
+    companion object {
+        private const val TAG = "WebViewCapture"
+        private const val BRIDGE_NAME = "BitdriftLogger"
+
+        /**
+         * Instruments a WebView to capture page load events, Core Web Vitals,
+         * and network requests.
+         *
+         * This method:
+         * - Wraps the existing WebViewClient to capture page load events
+         * - Enables JavaScript execution
+         * - Injects the Bitdrift JavaScript bridge at document start
+         * - Registers a native interface for receiving bridge messages
+         *
+         * @param webview The WebView to instrument
+         * @param logger Optional logger instance. If null, uses Capture.logger()
+         */
+        @SuppressLint("SetJavaScriptEnabled", "RequiresFeature")
+        @JvmStatic
+        @JvmOverloads
+        fun instrument(webview: WebView, logger: ILogger? = null) {
+            val effectiveLogger = logger ?: Capture.logger()
+            
+            // Wrap existing WebViewClient
+            val capture = if (WebViewFeature.isFeatureSupported(WebViewFeature.GET_WEB_VIEW_CLIENT)) {
+                val original = WebViewCompat.getWebViewClient(webview)
+                WebViewCapture(original, effectiveLogger)
+            } else {
+                WebViewCapture(WebViewClient(), effectiveLogger)
+            }
+            
+            webview.webViewClient = capture
+
+            // Enable JavaScript (required for bridge)
+            webview.settings.javaScriptEnabled = true
+
+            // Register JavaScript interface for receiving bridge messages
+            val bridgeHandler = WebViewBridgeHandler(effectiveLogger, capture)
+            webview.addJavascriptInterface(bridgeHandler, BRIDGE_NAME)
+
+            // Inject JavaScript at document start for early initialization
+            injectScript(webview, effectiveLogger)
+        }
+
+        @SuppressLint("RequiresFeature")
+        private fun injectScript(webview: WebView, logger: ILogger?) {
+            if (!WebViewFeature.isFeatureSupported(WebViewFeature.DOCUMENT_START_SCRIPT)) {
+                // Fall back to evaluateJavascript after page load on older WebViews
+                logger?.log(LogLevel.DEBUG, emptyMap()) {
+                    "WebView DOCUMENT_START_SCRIPT not supported, using fallback injection"
+                }
+                return
+            }
+
+            try {
+                val script = WebViewBridgeScript.SCRIPT
+                WebViewCompat.addDocumentStartJavaScript(
```

Comment:
wouldn't we be calling this _in addition_ to  `webview.addJavascriptInterface()` for webviews that support both?